tcms/telemetry/static/telemetry/js/testing/execution-trends.js

```diff
@@ -0,0 +1,95 @@
+$(document).ready(() => {
+    loadInitialProduct();
+    loadInitialTestPlans();
+
+    document.getElementById('id_product').onchange = () => {
+        update_version_select_from_product();
+        update_build_select_from_product(true);
+        updateTestPlanSelectFromProduct(drawChart);
+    };
+
+    document.getElementById('id_version').onchange = drawChart;
+    document.getElementById('id_build').onchange = drawChart;
+    document.getElementById('id_test_plan').onchange = drawChart;
+
+    $('#id_after').on('dp.change', drawChart);
+    $('#id_before').on('dp.change', drawChart);
+
+    drawChart();
+});
+
+function drawChart() {
+    const query = {};
+
+    const productId = $('#id_product').val();
+    if (productId) {
+        query['case__plan__product'] = productId;
+    }
+
+    const versionId = $('#id_version').val();
+    if (versionId) {
+        query['case__plan__product_version'] = versionId;
+    }
+
+    const buildId = $('#id_build').val();
+    if (buildId) {
+        query['build_id'] = buildId;
+    }
+
+    const testPlanId = $('#id_test_plan').val();
+    if (testPlanId) {
+        query['case__plan__plan_id'] = testPlanId;
+    }
+
+    const dateBefore = $('#id_before');
+    if (dateBefore.val()) {
+        query['close_date__lte'] = dateBefore.data('DateTimePicker').date().format('YYYY-MM-DD 23:59:59');
+    }
+
+    const dateAfter = $('#id_after');
+    if (dateAfter.val()) {
+        query['close_date__gte'] = dateAfter.data('DateTimePicker').date().format('YYYY-MM-DD 00:00:00');
+    }
+
+    jsonRPC('Testing.execution_trends', query, data => {
+        const chartData = [];
+
+        Object.entries(data.data_set).forEach(entry => {
+            chartData.push([entry[0], ...entry[1]]);
+        });
+
+        $('#chart > svg').remove();
+
+        const c3ChartDefaults = $().c3ChartDefaults();
+        const config = c3ChartDefaults.getDefaultAreaConfig();
+        config.axis = {
+            x: {
+                categories: data.categories,
+                type: 'category',
+                tick: {
+                    format: runId => `TR-${runId}`
```

Comment:
label must be horizontal, not vertical

Reply:
One option is to do this:
```javascript
config.axis = {
            x: {
                tick: {
                    format: runId => `TR-${runId}`,
                    fit: false,
                    multiline: false,
                },
            },
```
basically saying to DataTables that we want the X axis label to be on line line and be separated from the others. this is the result:
<img width="457" alt="Screenshot 2019-07-14 at 12 38 43" src="https://user-images.githubusercontent.com/18421997/61181854-5a007380-a634-11e9-98c6-7635921a1bb3.png">
however, in order to achieve that DataTables would just omit some of the labels in order to fit the other ones (see TR-1 label is missing on the picture, along with the other odd labels)

Are you ok with something like this? I could not find an option which would just make the whole column wider, just to fit the labels

Reply:
go for it

Reply:
That change is present in this PR. you can review it